internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -158,37 +264,75 @@ func (p *processRunner) build() (commandWrapper, error) {
 			return commandWrapper{}, err
 		}
 
-		cmd.Stderr = stderr
+		wrapper.stderr = stderr.Fd()
 
 		afterStartFuncs = append(afterStartFuncs, func() {
 			stderr.Close() //nolint:errcheck
 		})
 	} else {
-		cmd.Stderr = writer
+		wrapper.stderr = pw.Fd()
 	}
 
-	ctty, cttySet := p.opts.Ctty.Get()
-	if cttySet {
-		cmd.SysProcAttr = &syscall.SysProcAttr{
-			Setsid:  true,
-			Setctty: true,
-			Ctty:    ctty,
+	closeWriter = false
+
+	wrapper.launcher = launcher
+	wrapper.afterStart = func() {
+		for _, f := range afterStartFuncs {
+			f()
 		}
 	}
+	wrapper.afterTermination = closeLogging
+	wrapper.ctty = p.opts.Ctty
 
-	closeWriter = false
+	cgroupFdSupported := false
 
-	return commandWrapper{
-		cmd: cmd,
-		afterStart: func() {
-			for _, f := range afterStartFuncs {
-				f()
-			}
-		},
-		afterTermination: func() error {
-			return w.Close()
-		},
-	}, nil
+	platform, err := platform.CurrentPlatform()
+	if err == nil {
+		cgroupFdSupported = platform.Mode() != runtime.ModeContainer
+	}
+
+	// cgroupfd is more reliable, use it when possible
+	if cgroups.Mode() == cgroups.Unified && cgroupFdSupported && p.opts.UID == 0 {
```

Comment:
Instead of using `p.opts.UID == 0` we could chown the dashboard cgroup to be owned by dashboard's UID. Else it'll give EACCES on clone3 call

Reply:
I think either way is fine, we can certainly chown the cgroups if we know the UIDs of the owner (e.g. the dashboard).

Reply:
Will then chown, since we decided our preferred way is using cgroupfd when possible

Reply:
chown is not really sufficient